src/appleseed/renderer/modeling/project/projectfilereader.cpp

```diff
@@ -3153,6 +3153,28 @@ namespace
 
         return false;
     }
+
+    // returns filename of .appleseed file inside archive
+    // returns "" in case there are more than 1 .appleseed files
+    // "" means it's not a valid packed project
+    string get_project_name(const char* project_filepath)
+    {
+        const vector<string> appleseed_files = get_filenames_with_extension_from_zip(project_filepath, ".appleseed");
+        return appleseed_files.size() == 1 ? appleseed_files[0] : "";
+    }
+
+    string get_unpacked_project(
```

Comment:
You're right, but maybe the log message itself could be moved to the method, I think it would make sense. In that case it should probably be changed from

    "%s appears to be a packed project; unpacking to %s..."

to something like

    "Unpacking %s to %s..."